src/lib.rs

```diff
@@ -67,11 +65,23 @@ impl Default for Client {
         // TODO: make configurable in `Client::builder()`.
         connector.set_keepalive(Some(TCP_KEEPALIVE));
 
-        #[cfg(feature = "tls")]
-        let connector = HttpsConnector::new_with_connector({
-            connector.enforce_http(false);
-            connector
-        });
+        #[cfg(any(feature = "tls", feature = "rustls"))]
```

Comment:
Since `v0.12` contains breaking changes anyway, let's discuss the feature set here.

One way is to use the `reqwest` crate's approach here with `default-tls` (auto choice), `rustls-tls` and `native-tls` features. Sadly, `clickhouse::Client` doesn't have a builder now to also provide runtime override [request::ClientBuilder::use_native_tls](https://docs.rs/reqwest/latest/reqwest/struct.ClientBuilder.html#method.use_native_tls), but we can create a builder (even with backward compatibility) or follow the current approach and create the connector lazily.

Another way is to not enable `tls` by default at all and force users to enable one of these features. Sadly, we should choose the priority if both features are enabled. This approach can be extended to the first one later.

Reply:
I think I have a slight preference for not enabling any `tls`-related feature by default (and have done so), but I could try my hand at the reqwest approach if you decide you prefer that.

I also chose the `rustls` feature to take priority over `tls`, because I believe that rustls is much better. I'm happy to flip this, though, as it only matters in the case where both `rustls` and `tls` features are enabled.

Reply:
We can start with disable-by-default TLS and force users to decide what to use, at least read documentation.

Anyway, `default-tls` can be added in the future.

> I also chose the rustls feature to take priority over tls, because I believe that rustls is much better
I agree